adoc/chapters/programming_interface.adoc

```diff
@@ -19401,481 +19401,1161 @@ that all of the built-in functions fulfill the same requirements for both
 host and device.
 
 
-=== Description of the built-in types available for SYCL host and device
+=== Description of generic type names
 
-All of the OpenCL built-in types are available in the namespace
-[code]#sycl#. For the purposes of this document we use
-generic type names for describing sets of valid SYCL types. The
-generic type names themselves are not valid SYCL types, but they
-represent a set of valid types, as defined in
-<<table.gentypes>>. Each generic type within a section is
-comprised of a combination of scalar, SYCL [code]#vec# and/or [code]#marray# class
-specializations. The letters [code]#{n}# and [code]#{N}# define valid sizes for
-class specializations, where [code]#{n}# means 2,3,4,8,16 and [code]#{N}# means
-any positive value of size_t type. Note that any reference to the base type
-refers to the type of a scalar or the element type of a SYCL [code]#vec# or
-[code]#marray# specialization.
+The functions defined in sections <<sec:math-functions>> through
+<<sec::relational-functions>> are specified using a set of generic type names.
+These are not actual type names available to SYCL applications.  Instead, they
+are a shorthand used in this document to describe the set of overloads for
+these functions.  For example the [code]#fmax# function is specified as:
 
+[source]
+----
+genfloat fmax(genfloat x, genfloat y)
+----
+
+The generic type [code]#genfloat# is a stand-in for a set of a actual type
+names: [code]#float#, [code]#double#, etc.  To construct the actual overloads
+that are available, replace the generic type name with the name of each of
+these actual types:
+
+[source]
+----
+float fmax(float x, float y)
+double fmax(double x, double y)
+vec<float,1> fmax(vec<float,1> x, vec<float,1> y)
+vec<float,2> fmax(vec<float,2> x, vec<float,2> y)
+vec<float,3> fmax(vec<float,3> x, vec<float,3> y)
+vec<float,4> fmax(vec<float,4> x, vec<float,4> y)
+vec<float,8> fmax(vec<float,8> x, vec<float,8> y)
+vec<float,16> fmax(vec<float,16> x, vec<float,16> y)
+template<size_t N> marray<float,N> fmax(marray<float,N> x, marray<float,N> y)
+// etc.
```

Comment:
```suggestion
// ...
```

Reply:
Added in bd1229a146abf422525dd25563865087e0d2fa1a